### DISCUSSION TITLE
Intermittent failures to create user account

### DISCUSSION THREAD

**Discussion by johndwells** [verbb/formie#2095] [Q&A]
We have a Formie form set up as a "registration form", for new users to register with the site. This works flawlessly 99% of the time, but occasionally the User account is never created, but there's no indication as to why. We do have Google reCAPTCHA configured, but these submissions are not being flagged as spam.

If I go into the CMS and find the submission entry, and then manually re-trigger the User integration, it runs flawlessly.

Any advice on how we might track down the root cause of these intermittent failures?

**Comment by engram-design** [comment]
How do you run your queue? Are there any error logs that relate to the submissions that haven't run properly?

**Comment by johndwells** [comment]
@engram-design It's hosted on Servd, so it's done on a dedicated queue runner. I've downloaded the server logs, but didn't see anything obvious in the queue logs. The logs don't show evidence of _any_ formie processing though, does it only log something if it encounters an error?

Are all integrations handled by the queue? So no processing is done at the time of submission?

**Reply by engram-design** [reply]
Ah, so Servd's logs don't persist, so that makes it tricky to diagnose, as they're usually gone by the time you need to look at them.

All integrations use the queue by default, and yes - no processing is done at submission. But of course you can turn that off in Formie's settings, which might be an option for this project.

**Reply by johndwells** [reply]
@engram-design That's not actually the case, Servd keeps logs for 30 days.

Would there still be an advantage to disabling use of the queue for integrations?

This might be a big ask, but would you know the sorts of things that would cause the integration to (occasionally) fail, and silently at that? I would think that a validation error (e.g. user already exists) would be logged. Would I need to turn on devMode or another setting to get enough logging?

**Reply by engram-design** [reply]
Hmm, then there _should_ be some logs for Formie, although our plugins log slightly differently to Craft under the hood (separate files). Last time I checked with them, there were some issues with that.

If there's any sort of error within the integration, the queue job will fail, and it'll sit there in an error state. I'm not sure if Servd has any handling for dealing with errored queue jobs, but I wouldn't have thought so.

I may need to spin up a Servd site to verify the logging behaviour to get started with this one.

**Reply by johndwells** [reply]
> Hmm, then there should be some logs for Formie, although our plugins log slightly differently to Craft under the hood (separate files). Last time I checked with them, there were some issues with that.

Ah - so yeah by default Servd doesn't capture any special log files - but I just have to let them know what file(s) to save. Let me tell them to include `formie` log files. I'll see if it's retroactive, but I doubt it.
